lib/interceptor/cache.js

```diff
@@ -0,0 +1,148 @@
+'use strict'
+
+const CacheHandler = require('../handler/cache-handler')
+const MemoryCacheStore = require('../cache/memory-cache-store')
+const CacheRevalidationHandler = require('../handler/cache-revalidation-handler')
+const { UNSAFE_METHODS } = require('../util/cache.js')
+
+const AGE_HEADER = Buffer.from('age')
+
+/**
+ * @param {import('../../types/cache-interceptor.d.ts').default.CacheOptions | undefined} globalOpts
+ * @returns {import('../../types/dispatcher.d.ts').default.DispatcherComposeInterceptor}
+ */
+module.exports = globalOpts => {
+  if (!globalOpts) {
+    globalOpts = {}
+  }
+
+  if (globalOpts.store) {
+    for (const fn of ['createReadStream', 'createWriteStream', 'deleteByOrigin']) {
```

Comment:
Shouldn't `size` or `isFull` be included?

Reply:
`isFull` is checked right after, `size` isn't a part of the interface in favor of just having `isFull`